lib/more_core_extensions/core_ext/string/decimal_suffix.rb

```diff
@@ -1,16 +1,28 @@
+require 'bigdecimal'
+
 module MoreCoreExtensions
   module DecimalSI
-    DECIMAL_SUFFIXES = {"d" => 1e-1, "c" => 1e-2, "m" => 1e-3, "μ" => 1e-6, "n" => 1e-9, "p" => 1e-12, "f" => 1e-15,
-                        "a" => 1e-18, "h" => 1e2, "k" => 1e3, "M" => 1e6, "G" => 1e9, "T" => 1e12, "P" => 1e15,
-                        "E" => 1e18}.freeze
+    DECIMAL_SUFFIXES = {"d" => "e-1", "c" => "e-2", "m" => "e-3", "μ" => "e-6", "n" => "e-9", "p" => "e-12", "f" => "e-15",
+                        "a" => "e-18", "h" => "e2", "k" => "e3", "M" => "e6", "G" => "e9", "T" => "e12", "P" => "e15",
+                        "E" => "e18"}.freeze
+
     def decimal_si_to_f
       multiplier = DECIMAL_SUFFIXES[self[-1]]
       if multiplier
-        Float(self[0..-2]) * multiplier
+        Float(self[0..-2] + multiplier)
       else
         Float(self)
       end
     end
+
+    def decimal_si_to_bigdecimal
+      multiplier = DECIMAL_SUFFIXES[self[-1]]
+      if multiplier
```

Comment:
Can this conditional be extracted to a new private method and shared between the two methods?

Reply:
:+1: done